flight/targets/brainre1/fw/pios_board.c

```diff
@@ -0,0 +1,534 @@
+ * the HW definitions to be const and static to limit their
+ * scope.
+ *
+ * NOTE: THIS IS THE ONLY PLACE THAT SHOULD EVER INCLUDE THIS FILE
+ */
+
+#include "board_hw_defs.c"
+
+#include <pios.h>
+#include <pios_hal.h>
+#include <openpilot.h>
+#include <misc_math.h>
+#include <uavobjectsinit.h>
+#include "hwbrainre1.h"
+#include "flightbatterysettings.h"
+#include "flightstatus.h"
+#include "modulesettings.h"
+#include "manualcontrolsettings.h"
+#include "onscreendisplaysettings.h"
+
+
+#include "pios_ir_transponder.h"
+
+#if defined(PIOS_INCLUDE_BMI160)
+#include "pios_bmi160.h"
+
+static const struct pios_exti_cfg pios_exti_bmi160_cfg __exti_config = {
+	.vector = PIOS_BMI160_IRQHandler,
+	.line = EXTI_Line13,
+	.pin = {
+		.gpio = GPIOC,
+		.init = {
+			.GPIO_Pin = GPIO_Pin_13,
+			.GPIO_Speed = GPIO_Speed_2MHz,  // XXXX
+			.GPIO_Mode = GPIO_Mode_IN,
+			.GPIO_OType = GPIO_OType_OD,
+			.GPIO_PuPd = GPIO_PuPd_NOPULL,
+		},
+	},
+	.irq = {
+		.init = {
+			.NVIC_IRQChannel = EXTI15_10_IRQn,
+			.NVIC_IRQChannelPreemptionPriority = PIOS_IRQ_PRIO_MID,
+			.NVIC_IRQChannelSubPriority = 0,
+			.NVIC_IRQChannelCmd = ENABLE,
+		},
+	},
+	.exti = {
+		.init = {
+			.EXTI_Line = EXTI_Line13, // matches above GPIO pin
+			.EXTI_Mode = EXTI_Mode_Interrupt,
+			.EXTI_Trigger = EXTI_Trigger_Rising,
+			.EXTI_LineCmd = ENABLE,
+		},
+	},
+};
+
+static const struct pios_bmi160_cfg pios_bmi160_cfg = {
+	.exti_cfg = &pios_exti_bmi160_cfg,
+	.orientation = PIOS_BMI160_TOP_0DEG,
+	.odr = PIOS_BMI160_ODR_1600_Hz,
+	.acc_range = PIOS_BMI160_RANGE_8G,
+	.gyro_range = PIOS_BMI160_RANGE_2000DPS,
+	.temperature_interleaving = 50
+};
+#endif /* PIOS_INCLUDE_BMI160 */
+
+#if defined(PIOS_INCLUDE_FRSKY_RSSI)
+#include "pios_frsky_rssi_priv.h"
+#endif /* PIOS_INCLUDE_FRSKY_RSSI */
+
+bool external_mag_fail;
+
+uintptr_t pios_com_logging_id;
+uintptr_t pios_com_openlog_logging_id;
+uintptr_t pios_internal_adc_id = 0;
+uintptr_t pios_uavo_settings_fs_id;
+uintptr_t pios_waypoints_settings_fs_id;
+
+
+const uint8_t named_color_value[HWBRAINRE1_LEDCOLOR_GLOBAL_MAXOPTVAL + 1][3] = {
+	[HWBRAINRE1_LEDCOLOR_OFF]    = {0,   0,   0},
+	[HWBRAINRE1_LEDCOLOR_CUSTOM] = {0,   0,   0},
+	[HWBRAINRE1_LEDCOLOR_WHITE]  = {255, 255, 255},
+	[HWBRAINRE1_LEDCOLOR_RED]    = {255, 0,   0},
+	[HWBRAINRE1_LEDCOLOR_ORANGE] = {255, 69,   0},
+	[HWBRAINRE1_LEDCOLOR_YELLOW] = {255, 255, 0},
+	[HWBRAINRE1_LEDCOLOR_GREEN]  = {0,   255, 0},
+	[HWBRAINRE1_LEDCOLOR_AQUA]   = {0,   255, 255},
+	[HWBRAINRE1_LEDCOLOR_BLUE]   = {0,   0,   255},
+	[HWBRAINRE1_LEDCOLOR_PURPLE] = {255, 0,   255},
+};
+
+
+/**
+ * settingdUpdatedCb()
+ * Called when RE1 hardware settings are updated
+ */
+static void settingdUpdatedCb(UAVObjEvent * ev, void *ctx, void *obj, int len)
+{
+	(void) ev; (void) ctx; (void) len;
+
+	HwBrainRE1Data * hwre1data = (HwBrainRE1Data *)obj;
+
+	/* Set the colors of the RGB LEDs */
+	if (hwre1data->NumberOfLEDs > 0) {
+		uint8_t led_color[3];
```

Comment:
For future.. i prefer `= {};` or `= {0};` to memset below.  Easier for compiler to infer/elide.